src/Storages/PostgreSQL/PostgreSQLSettings.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <Core/Defines.h>
+#include <Core/BaseSettings.h>
+#include <Core/SettingsEnums.h>
+#include <Interpreters/Context_fwd.h>
+
+
+namespace Poco::Util
+{
+    class AbstractConfiguration;
+}
+
+
+namespace DB
+{
+class ASTStorage;
+class ASTSetQuery;
+
+#define LIST_OF_POSTGRESQL_SETTINGS(M, ALIAS) \
+    M(UInt64, postgresql_connection_pool_size, 16, "Size of connection pool (if all connections are in use, the query will wait until some connection will be freed).", 0) \
```

Comment:
I'd rather omit "postgresql" prefix here. Seems ok as `CREATE TABLE .. ENGINE = PostgreSQL(...) SETTINGS postgresql_connection_pool_size = n` never worked before and ok that in DB::Settings we leave it with "postgresql" prefix as those settings are common.